TheEthicalHackerCup/Assets/AttackMinigames/Ransomware/Scripts/Collectable.cs

```diff
@@ -0,0 +1,50 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class Collectable : MonoBehaviour
+{
+    [SerializeField]
+    private float timeToCollect = 3f;
+    [SerializeField]
+    private Sprite collectedSprite;
+
+    private float collectedPercent = 0f;
+    private bool isBeingCollected = false;
+    private bool isCollected = false;
+    private RansomManager manager;
+    private SpriteRenderer spriteRenderer;
+    
+
+    // Start is called before the first frame update
+    void Start()
+    {
+        manager = FindObjectOfType<RansomManager>();
+        spriteRenderer = GetComponent<SpriteRenderer>();
+        manager.FileCount++;
+    }
+
+    private void OnTriggerStay2D(Collider2D collision)
+    {
+        
+        var other = collision.gameObject;
+        isBeingCollected = other.gameObject.CompareTag("Player");
+        if (isBeingCollected && !isCollected)
+        {
+            collectedPercent += Time.deltaTime / timeToCollect;
```

Comment:
Consider making the collectables prefabs and then incorporating something to that indicates the collect percent.

Perhaps a text component: `21% Captured`
Perhaps rotate the sprite until it is upside down where it is captured
Perhaps an initially invisible lock sprite on top of the collectible that increases in opacity as it gets captured

My main concern is that players will be like "I touched it, why didn't I collect it?" if there is no feedback about collection progress

Reply:
todo: this